runtime/runtime/src/lib.rs

```diff
@@ -1752,129 +1752,51 @@ impl Runtime {
-                    state_update.trie.recorded_storage_size_upper_bound() > limit
-                })
-            {
-                break;
-            }
-
-            let queue_entry_key =
-                TrieKey::PromiseYieldTimeout { index: promise_yield_indices.first_index };
-
-            let queue_entry = get::<PromiseYieldTimeout>(state_update, &queue_entry_key)?
-                .ok_or_else(|| {
-                    StorageError::StorageInconsistentState(format!(
-                        "PromiseYield timeout queue entry #{} should be in the state",
-                        promise_yield_indices.first_index
-                    ))
-                })?;
-
-            // Queue entries are ordered by expires_at
-            if queue_entry.expires_at > apply_state.block_height {
-                break;
-            }
-
-            // Check if the yielded promise still needs to be resolved
-            let promise_yield_key = TrieKey::PromiseYieldReceipt {
-                receiver_id: queue_entry.account_id.clone(),
-                data_id: queue_entry.data_id,
-            };
-            if state_update.contains_key(&promise_yield_key)? {
-                let new_receipt_id = create_receipt_id_from_receipt_id(
-                    protocol_version,
-                    &queue_entry.data_id,
-                    &apply_state.prev_block_hash,
-                    &apply_state.block_hash,
-                    new_receipt_index,
-                );
-                new_receipt_index += 1;
-
-                // Create a PromiseResume receipt to resolve the timed-out yield.
-                let resume_receipt = Receipt::V0(ReceiptV0 {
-                    predecessor_id: queue_entry.account_id.clone(),
-                    receiver_id: queue_entry.account_id.clone(),
-                    receipt_id: new_receipt_id,
-                    receipt: ReceiptEnum::PromiseResume(DataReceipt {
-                        data_id: queue_entry.data_id,
-                        data: None,
-                    }),
-                });
-
-                // The receipt is destined for the local shard and will be placed in the outgoing
-                // receipts buffer. It is possible that there is already an outgoing receipt resolving
-                // this yield if `yield_resume` was invoked by some receipt which was processed in
-                // the current chunk. The ordering will be maintained because the receipts are
-                // destined for the same shard; the timeout will be processed second and discarded.
-                receipt_sink.forward_or_buffer_receipt(
-                    resume_receipt.clone(),
-                    apply_state,
-                    &mut state_update,
-                    processing_state.epoch_info_provider,
-                )?;
-                timeout_receipts.push(resume_receipt);
-            }
+        let promise_yield_result = resolve_timed_out_promise_yield_receipts(
```

Comment:
I saw the term "promise_yield_timeout" somewhere else in the codebase. Can you check if this is the convention and if so rename this to `resolve_promise_yield_timeouts`?

Reply:
Yes, the `promise_yield_timeout` name appears consistently in this context. I'll apply your suggestion.